app/lua/lbaselib.c

```diff
@@ -676,14 +674,12 @@ static void base_open (lua_State *L) {
   /* set global _G */
   lua_pushvalue(L, LUA_GLOBALSINDEX);
   lua_setglobal(L, "_G");
+
   /* open lib into global table */
-  luaL_register_light(L, "_G", base_funcs);
-#if LUA_OPTIMIZE_MEMORY > 0
-  lua_pushvalue(L, -1);
-  lua_setmetatable(L, -2);
-  lua_pushrotable(L, (void *)BASE_ROTABLE);
-  lua_setglobal(L, "__index");
-#endif
+  luaL_register_light(L, "_G", &((luaL_Reg) {0}));
```

Comment:
`luaL_register_light()` does multiple things.  It registers the library in the global table (`_G` itself in this case) and also registers a list of functions.  There aren't any functions to be registered here since these are in a ROTable for the baselib, so we either call `luaL_register_light()` with an empty list or clone and copy the subset of the functionality that we used inline.  Doing the call is simpler than cut and paste.